colab/colab_utils.py

```diff
@@ -0,0 +1,1019 @@
+      post_images.append(e.features.feature['post_image_png'].bytes_list.value[0])
+      labels.append(e.features.feature['label'].float_list.value[0])
+
+  labels_counter=dict(collections.Counter(labels_split))
+  map_value = {0: 'Undamaged/bad examples {}/{}'.format(int(len(labels)-sum(labels)),labels_counter[0]),
+               1: 'Damaged {}/{}'.format(int(sum(labels)),labels_counter[1])}
+  labels=list((pd.Series(labels)).map(map_value))
+  
+  images=[concat_caption_pilimage(pre_images[idx], post_images[idx]) for idx in range(len(pre_images))]
+
+  ipyplot.plot_class_tabs(images, labels,max_imgs_per_tab=max_examples, tabs_order=[map_value[1],map_value[0]], img_width=200)
+
+  return total_example_num
+
+
+## CLASS DEFINITION TRAINJOB
+
+def write_train_and_eval_launch_script(**args):
+
+  args['hyper_parameters_args']=''
+
+  submission_ending = '''
+source {python_env}; export GOOGLE_APPLICATION_CREDENTIALS={path_cred};
+python {path_skai}/src/launch_vertex_job.py \\
+  --location={cloud_region} \\
+  --project={cloud_project} \\
+  --job_type=train \\
+  --display_name={display_name_train} \\
+  --dataset_name={dataset_name} \\
+  --train_worker_machine_type=n1-highmem-16 \\
+  --train_docker_image_uri_path={train_docker_image_uri_path} \\
+  --service_account={service_account} \\
+  --train_dir={train_dir} \\
+  --train_unlabel_examples={train_unlabel_examples} \\
+  --train_label_examples={train_label_examples} \\
+  --test_examples={test_examples} & \\
+sleep 60 ; python {path_skai}/src/launch_vertex_job.py \\
+  --location={cloud_region} \\
+  --project={cloud_project} \\
+  --job_type=eval \\
+  --display_name={display_name_eval} \\
+  --dataset_name={dataset_name} \\
+  --eval_docker_image_uri_path={eval_docker_image_uri_path} \\
+  --service_account={service_account} \\
+  --train_dir={train_dir} \\
+  --train_unlabel_examples={train_unlabel_examples} \\
+  --train_label_examples={train_label_examples} \\
+  --test_examples={test_examples}'''.format(**args)
+  print(submission_ending)
+  with open(args['path_run'], 'w+') as file:
+    file.write(submission_ending)
+    
+def metrics(train_label_acc, train_label_auc, test_acc, test_auc, train_epoch,test_epoch, time):
+  html = """
+         <h2>Metrics (updated as training progresses {timestamp}):</h2>
+         <h3>Labeled Training Set, Epoch {train_epoch}</h3> 
+         <p>Accuracy: {train_label_acc}% | AUC: {train_label_auc}</p>
+         <h3>Test Set, Epoch {test_epoch}</h3>
+         <p>Accuracy: {test_acc}% | AUC: {test_auc}</p>
+        """.format(train_label_acc=train_label_acc, train_label_auc=train_label_auc,
+                   test_acc=test_acc, test_auc=test_auc,
+                   train_epoch=train_epoch,test_epoch=test_epoch,
+                   timestamp=time )
+  return HTML(html)
+
+def timestamp_to_datetime(timestamp):
+  return pd.to_datetime(timestamp)
+
+def update_job_id(job_id, train_job_id,eval_job_id):
```

Comment:
This function is really mysterious. I think it will be more clear just to get rid of it and move the code in the body to the place where it's called.

Reply:
change commit 6d5256b
Details: FYI the rationale here is to catch in the log the id for the training and evaluation job (in this order).